clients/src/main/resources/common/message/VotersRecord.json

```diff
@@ -0,0 +1,47 @@
+// Licensed to the Apache Software Foundation (ASF) under one or more
+// contributor license agreements.  See the NOTICE file distributed with
+// this work for additional information regarding copyright ownership.
+// The ASF licenses this file to You under the Apache License, Version 2.0
+// (the "License"); you may not use this file except in compliance with
+// the License.  You may obtain a copy of the License at
+//
+//    http://www.apache.org/licenses/LICENSE-2.0
+//
+// Unless required by applicable law or agreed to in writing, software
+// distributed under the License is distributed on an "AS IS" BASIS,
+// WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+// See the License for the specific language governing permissions and
+// limitations under the License.
+
+{
+  "type": "data",
+  "name": "VotersRecord",
+  "validVersions": "0",
+  "flexibleVersions": "0+",
+  "fields": [
+    { "name": "Version", "type": "int16", "versions": "0+",
+      "about": "The version of the voters record" },
+    { "name": "Voters", "type": "[]Voter", "versions": "0+", "fields": [
+      { "name": "VoterId", "type": "int32", "versions": "0+", "entityType": "brokerId",
+        "about": "The replica id of the voter in the topic partition" },
+      { "name": "VoterUuid", "type": "uuid", "versions": "0+",
+        "about": "The directory id of the voter in the topic partition" },
+      { "name": "Endpoints", "type": "[]Endpoint", "versions": "0+",
+        "about": "The endpoint that can be used to communicate with the voter", "fields": [
+        { "name": "Name", "type": "string", "versions": "0+", "mapKey": true,
+          "about": "The name of the endpoint" },
+        { "name": "Host", "type": "string", "versions": "0+",
+          "about": "The hostname" },
+        { "name": "Port", "type": "uint16", "versions": "0+",
```

Comment:
Yeah. @junrao and I had a discussion about this in the mailing list. The original design had the security protocol. I agreed to remove it because it is not strictly required since the local replica can look up the security protocol from its local configuration.

As you point out it would be useful to have this to catch misconfiguration between the replicas. What do you think @junrao should we add security protocol to all of the endpoint descriptions?

Reply:
@jsancio : Yes, adding the security protocol could potentially detect some, but not all misconfigurations. For example, SASL is one security protocol that supports different authentication mechanisms such as GSSAPI, OAuth, Plain, Scram, etc. If a user configures the listener with incorrect mechanism within SASL, just including the security protocol won't help. If want to solve the misconfiguration problem completely, should we include mechanism too? If you extend further, there could be other misconfigurations in login configurations for each specific mechanism too. How far should we go?

If we do include security protocol in the record, we probably need to revisit other RPCs like AddVoterRequest to see if security protocol needs to be included where listener name is used.